js/wp-seo-shortcode-plugin.js

```diff
@@ -0,0 +1,281 @@
+/* global console */
+/* global tinyMCE */
+/* global wpseoShortcodePluginL10n */
+/* global ajaxurl */
+/* global YoastSEO */
+/* global yoastShortcodePlugin */
+/* jshint -W097 */
+/* jshint -W003 */
+'use strict';
+
+/**
+ * The Yoast Shortcode plugin parses the shortcodes in a given piece of text. It analyzes multiple input fields for shortcodes which it will preload using AJAX.
+ *
+ * @constructor
+ */
+var YoastShortcodePlugin = function() {
+	this.registerTime = 0;
+	this.register( this.bindElementEvents );
+
+	this.unparsedShortcodes = [];
+	this.parsedShortcodes = [];
+
+	// The regex for matching shortcodes based on the available shortcode keywords.
+	this.keywordRegex = '(' + wpseoShortcodePluginL10n.wpseo_shortcode_tags.join('|') + ')';
+
+	this.loadShortcodes();
+};
+
+/* YOAST SEO CLIENT */
+
+/**
+ * Register the plugin with YoastSEO, retry for one second.
+ *
+ * @param {function} callback (optional) function to call when the plugin has been registered. This is useful for event bindings which rely on YoastSEO to be there.
+ */
+YoastShortcodePlugin.prototype.register = function( callback ) {
+	if ( typeof YoastSEO !== 'undefined' && typeof YoastSEO.app !== 'undefined' && typeof YoastSEO.app.plugins !== 'undefined' ) {
+		YoastSEO.app.plugins.register( 'YoastShortcodePlugin', { status: 'loading' } );
+		if ( typeof callback === 'function' ) {
+			callback();
+		}
+	}
+	else if ( this.registerTime < 1001 ) {
+		this.registerTime += 100;
+		setTimeout( function() {
+			yoastShortcodePlugin.register( callback );
+		}, 100 );
+	}
+	else {
+		console.error('Failed to register shortcode plugin with YoastSEO. YoastSEO is not available.');
+	}
+};
+
+/**
+ * Declares ready with YoastSEO.
+ */
+YoastShortcodePlugin.prototype.declareReady = function() {
+	YoastSEO.app.plugins.ready( 'YoastShortcodePlugin' );
+	yoastShortcodePlugin.registerModifications();
+};
+
+/**
+ * Declares reloaded with YoastSEO.
+ */
+YoastShortcodePlugin.prototype.declareReloaded = function() {
+	YoastSEO.app.plugins.reloaded( 'YoastShortcodePlugin' );
+};
+
+/**
+ * Registers the modifications for the content in which we want to replace shortcodes.
+ */
+YoastShortcodePlugin.prototype.registerModifications = function() {
+	YoastSEO.app.plugins.registerModification( 'title', this.replaceShortcodes, 'YoastShortcodePlugin' );
+	YoastSEO.app.plugins.registerModification( 'content', this.replaceShortcodes, 'YoastShortcodePlugin' );
+};
+
+/**
+ * The callback used to replace the shortcodes.
+ *
+ * @param {string} data
+ * @returns {string}
+ */
+YoastShortcodePlugin.prototype.replaceShortcodes = function( data ) {
+	var parsedShortcodes = yoastShortcodePlugin.parsedShortcodes;
```

Comment:
What is the reason to use `yoastShortcodePlugin.parsedShortcodes` here instead of `this.parsedShortcodes`?